R/HintikkaXOData.R

```diff
@@ -35,42 +33,61 @@
 #' \emph{International Journal of Environmental Research and Public Health}
 #' 18(8):4049 \url{https://doi.org/10.3390/ijerph18084049}
 #'
+#' @importFrom SummarizedExperiment rowData
+#' @importFrom SummarizedExperiment rowData<-
+#' 
 #' @name HintikkaXOData
-#' @importFrom SingleCellExperiment altExp
-#' @importFrom SingleCellExperiment altExp<-
 #' @export
 #'
 #' @examples
-#' # tse <- HintikkaXOData()
+#' # Retrieve the MAE data
+#' mae <- HintikkaXOData()
+#'
+#' # List the experiments in this MultiAssayExperiment object
+#' print(experiments(mae))
+#'
+#' # colData for this MAE data object
+#' colData(mae)
+#'
+#' # metabolite assay data
+#' # assays(mae[["metabolites"]])$assay
+#'
+#' # biomarker assay data
+#' # assays(mae[["biomarkers"]])$assay
+# '
+#' # microbiome assay counts
+#' # assays(mae[["microbiome"]])$counts
 #'
+#' # microbiome rowData
+#' rowData(mae[["microbiome"]])
+#' 
 HintikkaXOData <- function() {
 
-    # Circumvent warnings
-    meta_cecum <- otu_cecum <- tax <- NULL
-
-    dataset <- "3.14/hintikka-xo"
-    se <- .create_se(dataset,
-                    assays = c("counts"),
-                    has.rowdata = TRUE,
-                    has.coldata = TRUE)
-
-    tse <- SummarizedExperiment(assays = list(counts = otu_cecum),
-                        colData = meta_cecum,
-                        rowData = tax)
+    mae <- .create_mae("3.14/hintikka-xo",
+    
+        types = list(microbiome  = list("TSE" = c("counts")),
+                    metabolites = list("SE"  = NULL),
+                    biomarkers  = list("SE"  = NULL)
+                ),
+ 
+        coldata = TRUE,
+        samplemap = FALSE,
```

Comment:
This is related to the discussion whether assay name has to be specified, like "metabolites_abundances.rds" instead of "metabolites.rds". Both solutions are possible, we could try to add documentation on this when situation allows.

If the assay name must be specified in the file name, then the data files need to be updated to ExperimentHub before this can work.

Supporting the option of not providing assay name as part of the file name is one possible solution (implemented now in this PR although not thoroughly tested). The advantage is that it can (i) simplify file naming (metabolites.rds is not easily confused with rowdata.rds or coldata.rds anyway..) and (ii) the assay name in such cases is standardized (e.g. always "assay" or some other term that we can agree). The latter point could be somewhat useful. The disadvantage is that there would be two different naming conventions. 

Requiring assay name as part of tile name works as well, but sometimes the assay naming is somewhat arbitrary; "counts" does not work for many cases, then it should be "abundance"/"abundances", "signal"/"signals", "assay"/"assays" etc. and there may be many varying conventions even with otherwise fully analogous data sources.

I can live with either solution, if there are reasons to prefer one of these over the other.

Reply:
> Supporting the option of not providing assay name as part of the file name is one possible solution (implemented now in this PR although not thoroughly tested). The advantage is that it can (i) simplify file naming (metabolites.rds is not easily confused with rowdata.rds or coldata.rds anyway..) and (ii) the assay name in such cases is standardized (e.g. always "assay" or some other term that we can agree). The latter point could be somewhat useful. The disadvantage is that there would be two different naming conventions.

What about two assays without a name? Wouldn't  work. 

The system was established requiring an assay name for that reason, to provide the most specific return value and not leaving anything for the subject of code. The system is also used by other packages, which makes it in my point of view as good practice.

In addition your PR is not stringent in this. In your metadata you have specified no assay names, but in the script preparing the data you have.

I think it would be better to update the data on ExperimentHub

Reply:
Yes, the rule in that case would be to specify the name always if there are more than one assay. But I also agree that this is less clear.

Following best practices is good and there are benefits in requiring assay name. Ok let's do so.

Yes sorry PR was not fully coherent; in part because I was still indecisive about this, now we can complete.

Ok, let us update this to ExperimentHub. First waiting for the other data set to complete as well.